lib/wire_structs.ml

```diff
@@ -45,6 +45,30 @@ module Ipv4_wire = struct
     | `UDP    -> 17
 end
 
+module Arpv4_wire = struct
+
+  cstruct arp {
+    uint8_t dst[6];
+    uint8_t src[6];
+    uint16_t ethertype;
+    uint16_t htype;
+    uint16_t ptype;
+    uint8_t hlen;
+    uint8_t plen;
+    uint16_t op;
+    uint8_t sha[6];
+    uint32_t spa;
+    uint8_t tha[6];
+    uint32_t tpa
+  } as big_endian
+
+  cenum op {
+    Op_request = 1;
+    Op_reply
+    } as uint16_t
+
+end
```

Comment:
In fact, pace previous discussions about pcap, shouldn't the wire formats
be placed into their own opam package for separate installation?

(ISTR someone saying that this could be done without splitting the repo via
opam.<package> or something...?)

On 6 May 2015 at 06:13, Thomas Gazagnaire <notifications@github.com> wrote:

> In lib/wire_structs.ml
> <https://github.com/mirage/mirage-tcpip/pull/134#discussion_r29756251>:
>
> > +    uint16_t ptype;
> > +    uint8_t hlen;
> > +    uint8_t plen;
> > +    uint16_t op;
> > +    uint8_t sha[6];
> > +    uint32_t spa;
> > +    uint8_t tha[6];
> > +    uint32_t tpa
> > +  } as big_endian
> > +
> > +  cenum op {
> > +    Op_request = 1;
> > +    Op_reply
> > +    } as uint16_t
> > +
> > +end
>
> I'm in favour of breaking the consistency of the library and put ARP wire
> format directly in arp.ml (or in arp_wire.ml).
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/mirage/mirage-tcpip/pull/134/files#r29756251>.
>



-- 
Richard Mortier
mort@cantab.net